### ISSUE TITLE
System.ComponentModel.Win32Exception

### ISSUE THREAD

**Issue by finwen** [EDDiscovery/EDDiscovery#47]
processFilename = processes[0].MainModule.FileName;

Cant access 64 bit programs.

Solve by  making EDDiscover 64 bits maybee?

**Comment by gregmalcolm** [comment]
I take it you mean have a 32 bit and 64 bit offering and let the installer
choose? Some folks play elite on 32 bit OS.

I'll have a looksie in a bit to see if If I can find any other solutions.

**Comment by finwen** [comment]
Via process id  we can do

private string GetMainModuleFilepath(int processId)
{
    string wmiQueryString = "SELECT ProcessId, ExecutablePath FROM Win32_Process WHERE ProcessId = " + processId;
    using (var searcher = new ManagementObjectSearcher(wmiQueryString))
    {
        using (var results = searcher.Get())
        {
            ManagementObject mo = results.Cast<ManagementObject>().FirstOrDefault();
            if (mo != null)
            {
                return (string)mo["ExecutablePath"];
            }
        }
    }
    return null;
}

**Comment by gregmalcolm** [comment]
That sounds promising! It's been a long time since I've had to do this, but
that sounds like the kind of trick involved. Though I've done this more
with VB6 and C++ based apps.

**Comment by robbyxp1** [comment]
Hi,

Commander Robby here (Robbyp on the forums).

I just downloaded it and compiled it, with some idea to see if I can help you in some way in the future ;-)

I'm having the same problem when I compiled it, just about to write a note on the forum then remembered I can report it this way.

Looking at the code, EDDirectory is only used in conjunction with editign AppConfig.xml, right? So it does not matter if you manually update AppConfig.xml.

There is a typo in line 998 of EDDiscoveryForm.cs, "Remeber".

I tried a 64 bit build, but it gives:

1>------ Build started: Project: EDDiscovery, Configuration: Debug Any CPU ------
1>C:\Code\EDDiscovery-master\EDDiscovery\Trilateration\Trilateration.cs(1,7,1,13): error CS0246: The type or namespace name 'Noesis' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Code\EDDiscovery-master\EDDiscovery\Trilateration\Trilateration.cs(781,17,781,34): error CS0246: The type or namespace name 'JavascriptContext' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Code\EDDiscovery-master\EDDiscovery\DB\VisitedSystemsClass.cs(69,21,69,27): warning CS0108: 'VisitedSystemsClass.Update()' hides inherited member 'VisitedSystemsClass.Update()'. Use the new keyword if hiding was intended.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

I've not done any more digging into that.

Cheers
